src/primitive/cuboid.rs

```diff
@@ -0,0 +1,222 @@
+use cgmath::{BaseFloat, Point3, Vector3};
+use cgmath::prelude::*;
+
+use {Aabb3, Ray3};
+use prelude::*;
+
+/// Cuboid primitive.
+///
+/// Have a cached set of corner points to speed up computation.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "eders", derive(Serialize, Deserialize))]
+pub struct Cuboid<S> {
+    /// Dimensions of the box
+    pub dim: Vector3<S>,
```

Comment:
this one seems unsafe. User is able to change `dim` without updating the half-dimensions or corners